django/test/testcases.py

```diff
@@ -150,6 +150,46 @@ def __call__(self):
         )
 
 
+class _AssertSignalSentContext(object):
+    def __init__(self, test_case, signal, required_kwargs=None):
+        self.test_case = test_case
+        self.signal = signal
+        self.required_kwargs = required_kwargs
+        self.signal_sent = False
+        self.received_kwargs = {}
```

Comment:
While `_AssertSignalSentContext` is an implementation detail of `assertSignalSent` and should be considered a private API I think it would be preferable to initialize `signal_sent` and `received_kwargs` in `__enter__()` as this  allows instances to be reused.